cypress/integration/rich-text/RichTextEditor.Tracking.spec.ts

```diff
@@ -0,0 +1,627 @@
+        ]);
+      });
+
+      it(`tracks ${label} (${type}) via hotkeys ${shortcut}`, () => {
+        richText.editor.click().type(shortcut).type(shortcut);
+
+        richText.expectTrackingValue([
+          insert('shortcut', { nodeType: type }),
+          remove('shortcut', { nodeType: type }),
+        ]);
+      });
+    });
+  });
+
+  describe('Quote', () => {
+    const methods: [string, string, () => void][] = [
+      [
+        'using the toolbar',
+        'toolbar-icon',
+        () => {
+          richText.toolbar.quote.click();
+        },
+      ],
+      [
+        'using hotkey (mod+shift+1)',
+        'shortcut',
+        () => {
+          richText.editor.type(`{${mod}}{shift}1`);
+        },
+      ],
+    ];
+
+    for (const [scenario, origin, toggleQuote] of methods) {
+      it(`tracks blockquote ${scenario}`, () => {
+        richText.editor.click();
+
+        toggleQuote();
+        toggleQuote();
+
+        richText.expectTrackingValue([
+          insert(origin, { nodeType: BLOCKS.QUOTE }),
+          remove(origin, { nodeType: BLOCKS.QUOTE }),
+        ]);
+      });
+    }
+  });
+
+  describe('Tables', () => {
+    const insertTable = () => {
+      richText.editor.click();
+      richText.toolbar.table.click();
+      return richText.editor;
+    };
+    const insertTableWithExampleData = () => {
+      insertTable()
+        .type('foo')
+        .type('{rightarrow}')
+        .type('bar')
+        .type('{rightarrow}')
+        .type('baz')
+        .type('{rightarrow}')
+        .type('quux');
+    };
+
+    const insertTableAction = () => action('insertTable', 'toolbar-icon');
+    const removeTableAction = (payload) => action('removeTable', 'viewport-interaction', payload);
+    const insertTableRowAction = (payload) =>
+      action('insertTableRow', 'viewport-interaction', payload);
+    const insertTableColumnAction = (payload) =>
+      action('insertTableColumn', 'viewport-interaction', payload);
+    const removeTableColumnAction = (payload) =>
+      action('removeTableColumn', 'viewport-interaction', payload);
+    const removeTableRowAction = (payload) =>
+      action('removeTableRow', 'viewport-interaction', payload);
+
+    it('tracks insert table', () => {
+      insertTable();
+
+      richText.expectTrackingValue([insertTableAction()]);
+    });
+
+    describe('Table Actions', () => {
+      const findAction = (action: string) => {
+        cy.findByTestId('cf-table-actions-button').click();
+        return cy.findByText(action);
+      };
+
+      const doAction = (action: string) => {
+        findAction(action).click({ force: true });
+      };
+
+      beforeEach(() => {
+        insertTableWithExampleData();
+      });
+
+      it('adds row above', () => {
+        doAction('Add row above');
+
+        richText.expectTrackingValue([
+          insertTableAction(),
+          insertTableRowAction({
+            tableSize: {
+              numColumns: 2,
+              numRows: 2,
```

Comment:
This doesn't seem right, the default table when we insert is 2x2, after inserting another row I was expecting 3x2. Don't we count table headers?

Reply:
The same applies to other tests below

Reply:
We get the table size before applying the action. I don't know why we do that, we can discuss it with the team.